src/main/java/htsjdk/samtools/seekablestream/SeekableStreamFactory.java

```diff
@@ -61,9 +66,28 @@ public static ISeekableStreamFactory getInstance(){
      * Does this path point to a regular file on disk and not something like a URL?
      * @param path the path to test
      * @return true if the path is to a file on disk
+     * @deprecated this method is simplistic and no longer particularly useful since IOPath allows similar access to
+     * various non-file data sources, internal use has been replaced with {@link #isSpecialCase(String)}
      */
+    @Deprecated
     public static boolean isFilePath(final String path) {
-        return ! ( path.startsWith("http:") || path.startsWith("https:") || path.startsWith("ftp:") );
+        return !isSpecialCaseUrlType(path);
+    }
+
+    /**
+     * is this path being handled by one of the legacy SeekableStream types (http(s) / ftp)
+     *
+     * @param path a path to check
+     * @return if the path is not being handled by a FileSystemProvider and it can be read by legacy streams
+     */
+    public static boolean isSpecialCase(final String path){
+        return !new HtsPath(path).isPath()  //if we have a provider for it that's what we'll use
```

Comment:
Maybe break out out a `hasNIOPlugin()` (or similar) method for the `new HtsPath(path).isPath()` everywhere, for clarity?

Reply:
Or should you be using `hasFileSystemProvider()` instead?